strawberryfields/program.py

```diff
@@ -765,49 +696,99 @@ def compile(self, backend='fock', **kwargs):
         Returns:
             Program: compiled program
         """
-        db = backend_database[backend]
+        if backend in specs.backend_databases:
```

Comment:
do we want to go with `database`? might confuse people that we have a relational DB or something?

Reply:
How about `backend_specs`?

Reply:
Done